crates/ruff_python_literal/src/format.rs

```diff
@@ -282,17 +314,68 @@ fn parse_precision(text: &str) -> Result<(Option<usize>, &str), FormatSpecError>
     })
 }
 
+/// Parses a format part within a format spec
+fn parse_nested_format_part<'a>(
+    parts: &'a RefCell<Vec<FormatPart>>,
+    text: &'a str,
+) -> Result<&'a str, FormatSpecError> {
+    let mut parts = parts.borrow_mut();
+    let mut end_bracket_pos = None;
+    let mut left = String::new();
```

Comment:
Nit: move these below the `if text.is_empty()`, to make clear that they're only used in the `else` case?